src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -2993,7 +2993,8 @@ public void WriteLargeKeyValue(bool formatted, bool skipValidation)
 
         private static void WriteTooLargeHelper(JsonWriterState state, ReadOnlySpan<byte> key, ReadOnlySpan<byte> value, bool noThrow = false)
         {
-            var output = new ArrayBufferWriter(1024);
+            // Resizing is too slow, even for outerloop tests, so initialize to a large output size up front.
+            var output = new ArrayBufferWriter(noThrow ? 40_000_000 : 1024);
```

Comment:
This was the crux of the change to resolve https://github.com/dotnet/corefx/issues/35724.

Reply:
Was it the resizing or the exception that caused issues?

Reply:
It was the resizing (i.e. the case when we don't throw and successfully write a large key/value pair - 3 MB each). There is no exception in that case.